frontend/src/ui/Access.tsx

```diff
@@ -0,0 +1,698 @@
+
+        const initialOptions = allOptions.filter(
+            item => !initialSelections.some(elem => elem.value === item.value)
+        );
+
+        // The ACL might not explicitly include admin, but since we still want to show
+        // the admin entry when logged in as admin, the item needs to be added manually.
+        if (kind === "User" && !initialSelections.some(
+            selection => selection.label === "Administrator"
+        )) {
+            initialSelections.splice(0, 0, defaultDummyOption);
+        }
+
+        const [selection, setSelection] = useState<MultiValue<Option>>(initialSelections);
+        const [options, setOptions] = useState<MultiValue<Option>>(initialOptions);
+
+        useImperativeHandle(ref, () => ({
+            getSelection: () => selection,
+            reset: () => {
+                setSelection(initialSelections);
+                setOptions(initialOptions);
+            },
+        }));
+
+
+        const remove = (item: Option) => {
+            const filterItem = (items: MultiValue<Option>) => items.filter(
+                option => option.value !== item.value
+            );
+
+            setSelection(prev => filterItem(prev));
+            setOptions(prev => allOptions.some(option => option.value === item.value)
+                ? allOptions.filter(entry => !filterItem(selection)
+                    .some(option => entry.value === option.value))
+                : [...prev, item]);
+
+            setRoleSelections({
+                readRoles: roleSelections.readRoles.filter(role => item.value !== role),
+                writeRoles: roleSelections.writeRoles.filter(role => item.value !== role),
+            });
+        };
+
+        const handleCreate = (inputValue: string) => {
+            if (!inputValue.startsWith("ROLE_")) {
+                return;
+            }
+            const newRole: Option = {
+                value: inputValue,
+                label: formatUnknownRole(inputValue),
+            };
+            setSelection(prev => [...prev, newRole]);
+
+            setRoleSelections({
+                ...roleSelections,
+                readRoles: [...roleSelections.readRoles, inputValue],
+            });
+        };
+
+        const handleChange = (choice: MultiValue<Option>) => {
+            const newRoles = choice
+                .filter(option => !selection.includes(option))
+                .map(option => option.value);
+
+            setSelection([...choice].sort(roleComparator));
+            setOptions(prev => prev.filter(
+                option => !choice.some(opt => opt.value === option.value)
+            ));
+            setRoleSelections({
+                ...roleSelections,
+                readRoles: [...roleSelections.readRoles, ...newRoles],
+            });
+        };
+
+        const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+            event.preventDefault();
+
+            if (kind === "User") {
```

Comment:
Should the `preventDefault` maybe go into the `if`? I can't paste anything into the groups select. In fact, can you also check whether the clipboard data contains `\n` and if not, we shouldn't prevent the default.

Reply:
> can you also check whether the clipboard data contains \n and if not, we shouldn't prevent the default

That would mean that users can't paste in a single name, wouldn't it? I'm thinking that we might want to allow that.

Reply:
In fact, maybe we even want to allow pasting in group names on the left side... though that's probably not a common use case.

Reply:
Yes, but then the name is just in the search box and can be added by just pressing enter, no? And since it's just a single name, pressing enter once doesn't seem too bad to me.

And I wouldn't add pasting group names, until someone asks for it.

Reply:
Right, that's ok then. I'll add the check for `\n`.